### ISSUE TITLE
[BUG] test-ssl-new fails on NonStop x86 (3.0 HEAD)

### ISSUE THREAD

**Issue by rsbeckerca** [openssl/openssl#18447]
The overnight build/test of `openssl-3.0` failed on NonStop x86 (Big Endian) in the `80-test_ssl_new`. This was passing at commit 6537beb6805eb49a45b3a41fdf093f934c6df2ad, which actually does not make much sense given the change bd4ed81faa3469fddecd273cafa49bd9996d4a88. This may be timing related so might new failed at other commits. Certificate expired messages are inside the whole log. There may be issues in the test that only kicked in on Jun 1st. There are 19 failures in various sub-tests. Here is an example of one of them:
[test-ssl-new.log](https://github.com/openssl/openssl/files/8814082/test-ssl-new.log)

```
# Subtest: Test configuration 12-ct.cnf
    1..9
../../util/wrap.pl perl -I ../../util/perl ../../test/generate_ssl_tests.pl ../../test/ssl-tests/12-ct.cnf.in none > 12-ct.cnf.none => 0
    ok 1 - Getting output from generate_ssl_tests.pl.
    ok 2 - Comparing generated 12-ct.cnf.none with ../../test/ssl-tests/12-ct.cnf.
        # Subtest: ../../test/ssl_test
        1..1
            # Subtest: test_handshake
            1..6
            ok 1 - iteration 1
            # ERROR: (int) 'result->result == test_ctx->expected_result' failed @ /home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.0_Monitor/test/ssl_test.c:36
            # [2] compared to [0]
            # INFO:  @ /home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.0_Monitor/test/ssl_test.c:39
            # ExpectedResult mismatch: expected Success, got ClientFail.
            # 00000000:error:0A000415:SSL routines:ssl3_read_bytes:sslv3 alert certificate expired:/home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.0_Monitor/ssl/record/rec_layer_s3.c:1584:SSL alert number 45
            # OPENSSL_TEST_RAND_ORDER=1654081534
            not ok 2 - iteration 2
            ok 3 - iteration 3
            # ERROR: (int) 'result->result == test_ctx->expected_result' failed @ /home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.0_Monitor/test/ssl_test.c:36
            # [2] compared to [0]
            # INFO:  @ /home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.0_Monitor/test/ssl_test.c:39
            # ExpectedResult mismatch: expected Success, got ClientFail.
            # 00000000:error:0A000415:SSL routines:ssl3_read_bytes:sslv3 alert certificate expired:/home/ituglib/randall/jenkins/.jenkins/workspace/OpenSSL-3.0_Monitor/ssl/record
```
Please let me know how I can help further in diagnosing this issue.

**Comment by mattcaswell** [comment]
Probably this a dupe of #15179. Fix is in flight in #18444.

**Comment by rsbeckerca** [comment]
Testing now

**Comment by rsbeckerca** [comment]
The build is now failing:
```
c99: running /usr/cmplr/ccomfe -Z8 -ZBL --c99 -ZO2 -Zg3 -ZS/tmp/t469762153ADFBCG.T -D_XOPEN_SOURCE -ZR8 -m -Zf -ZT -D_TANDEM_SOURCE -ZNW203 -ZNW220 -ZNW272 -ZNW734 -ZNW770 -ZNW1506 -Zz -Iinclude -Iapps/include -I/usr/include -DOPENSSL_BUILDING_OPENSSL -DOPENSSL_VPROC= -D_XOPEN_SOURCE -D_XOPEN_SOURCE_EXTENDED=1 -D_TANDEM_SOURCE -DB_ENDIAN -D_TANDEM_ARCH=3 -DNDEBUG test/threadstest.c
      return thread_run_test(&test_lib_ctx_load_config_worker,
             ^
"/home/ituglib/randall/openssl-3.0/test/threadstest.c", line 566: error(114):
          identifier "thread_run_test" is undefined

                             MAXIMUM_THREADS, &test_lib_ctx_load_config_worker,
                             ^
"/home/ituglib/randall/openssl-3.0/test/threadstest.c", line 567: error(114):
          identifier "MAXIMUM_THREADS" is undefined

                             1, default_provider);
                                ^
"/home/ituglib/randall/openssl-3.0/test/threadstest.c", line 568: error(114):
          identifier "default_provider" is undefined

3 errors detected in the compilation of "test/threadstest.c".
c99: /usr/cmplr/ccomfe exited, returning 2.
```

**Comment by t8m** [comment]
> @t8m The build is now failing:

My fault from mistaken cherry-pick to 3.0 branch. Fix in #18451

**Comment by rsbeckerca** [comment]
> > @t8m The build is now failing:
> 
> My fault from mistaken cherry-pick to 3.0 branch. Fix in #18451

Looks good to me now.
